### PR TITLE
Use manpage subsections for GUI applications, and fix everything that is necessary to make this work on the website

### PR DESCRIPTION
**Pull request by kleinesfilmroellchen**
First of all, this PR fixes basically everything relating to links (and images) in subsections on the website.

Second, this PR introduces and improves some manpage checks (binary link names and old-style inter-manpage links), and applies necessary fixes.

Third, this PR moves all GUI applications into man1/Applications, which declutters the directory significantly and also introduces a logical separation between POSIX and command line programs versus the GUI applications.

### CODE REVIEW

File: Meta/build-manpages-website.sh [hunk 1]
```diff
@@ -106,7 +106,7 @@ pandoc -f gfm -t html5 -s \
 echo 'Copying images'
 rsync -a Meta/Websites/man.serenityos.org/banner.png output/ &
 rsync -a Base/usr/share/man/man7/LibDSP_classes.svg output/ &
-rsync -a Base/usr/share/man/man1/*.png output/ &
+find Base/usr/share/man/ -iname '*.png' -exec rsync -a {} output/ \; &
```
Comment (linusg) [inline Meta/build-manpages-website.sh:109]: what's `\;` for?

**Review by linusg — COMMENTED** [review]
> though it might in rare circumstances trigger a false positive.

How does one make CI pass in those cases?

> We do not want to move POSIX utilities into subsections since they have standard names

Still, what prevents us from doing the same for `man1/Utilities`?

**Comment by kleinesfilmroellchen** [comment]
> How does one make CI pass in those cases?

We have to create exceptions, if it does actually come up. There's already plenty of temporary and permanent exceptions to link checking.

> Still, what prevents us from doing the same for `man1/Utilities`?

You can't write something like `man cat` anymore, since `cat` is not a page, `Utilities/cat` is the page. The names of POSIX utility and function manpages are standardizes, as I am saying, and modifying that will break expectations with how a POSIX manpage system works.

File: Meta/build-manpages-website.sh (hunk 1 above)
Comment (kleinesfilmroellchen) [reply Meta/build-manpages-website.sh:109]: I think it terminates `-exec`; find does not run without it.

**Comment by linusg** [comment]
> You can't write something like `man cat` anymore, since cat is not a page, `Utilities/cat` is the page

That's ridiculous, `man foo` should still find `foo` in a subsection if it's unique. I don't want to type `man (or Help) Applications/Whatever` for this.

**Comment by kleinesfilmroellchen** [comment]
@linusg okay, but that necessitates further changes. How about "this PR" -> "more query support in LibManual" -> "Utilities subsection"?

**Comment by linusg** [comment]
Sure :)

**Pull request merged**